skrf/network.py

```diff
@@ -1642,7 +1682,33 @@ def copy_subset(self, key):
         except(AttributeError):
             ntwk.port_names = None
         return ntwk
-
+    
+#    @classmethod
+    def set_noise_a(self, noise_freq=None, nfmin_db=0, gamma_opt=0, rn=1 ) :
+          '''
+          sets the "A" (ie cascade) representation of the correlation matrix, based on the 
+          noise frequency and input parameters. 
+          '''
+          sh_fr = noise_freq.f.shape 
+          nfmin_db = npy.broadcast_to(npy.atleast_1d(nfmin_db), sh_fr)
+          gamma_opt = npy.broadcast_to(npy.atleast_1d(gamma_opt), sh_fr)
+          rn = npy.broadcast_to(npy.atleast_1d(rn), sh_fr)
+          
+              
+          nf_min = npy.power(10., nfmin_db/10.)
+          # TODO maybe interpolate z0 as above
+          y_opt = 1./(self.z0[0, 0] * (1. + gamma_opt)/(1. - gamma_opt))
+          noise = 4.*K_BOLTZMANN*T0*npy.array(
+                [[rn, (nf_min-1.)/2. - rn*npy.conj(y_opt)],
+                [(nf_min-1.)/2. - rn*y_opt, npy.square(npy.absolute(y_opt)) * rn]]
+              )
+          self.noise = noise.swapaxes(0, 2).swapaxes(1, 2)
+          self.noise_freq = noise_freq
+          xx=1
```

Comment:
Delete me

Reply:
Hi Tom,
Thank you for your comments and corrections. 
I corrected the extra line. 
The noise parameters are traditionally connected to port 1. 
and the noise calculation is performed only in one direction in the "cascade" operation. So does the .inv operation. 
ADS is a bit more clever with that respect , but there are still constraints. I would leave it as it is and put a note in the doc.